src/main/java/software/amazon/documentdb/jdbc/DocumentDbDatabaseMetaData.java

```diff
@@ -639,9 +657,393 @@ private void addImportedKey(final List<List<Object>> metaData,
+                                BASE_10 // NUM_PREC_RADIX
+                        )),
+                        new ArrayList<>(Arrays.asList(
+                                JdbcType.NCHAR.name(), // TYPE_NAME
+                                JdbcType.NCHAR.getJdbcType(), // DATA_TYPE
+                                TYPE_COLUMN_SIZE_MAP.get(JdbcType.NCHAR), // PRECISION
+                                "'", // LITERAL_PREFIX
+                                "'", // LITERAL_SUFFIX
+                                null, // CREATE_PARAMS
+                                ResultSetMetaData.columnNullable, // NULLABLE
+                                true, // CASE_SENSITIVE
+                                DatabaseMetaData.typeSearchable, // SEARCHABLE
+                                true, // UNSIGNED_ATTRIBUTE
+                                false, // FIXED_PREC_SCALE
+                                false, // AUTO_INCREMENT
+                                null, // LOCAL_TYPE_NAME
+                                0, // MINIMUM_SCALE
+                                0, // MAXIMUM_SCALE
+                                null, // SQL_DATA_TYPE (unused)
+                                null, // SQL_DATETIME_SUB (unused)
+                                BASE_10 // NUM_PREC_RADIX
+                        )),
+                        new ArrayList<>(Arrays.asList(
+                                JdbcType.NVARCHAR.name(), // TYPE_NAME
+                                JdbcType.NVARCHAR.getJdbcType(), // DATA_TYPE
+                                TYPE_COLUMN_SIZE_MAP.get(JdbcType.NVARCHAR), // PRECISION
+                                "'", // LITERAL_PREFIX
+                                "'", // LITERAL_SUFFIX
+                                null, // CREATE_PARAMS
+                                ResultSetMetaData.columnNullable, // NULLABLE
+                                true, // CASE_SENSITIVE
+                                DatabaseMetaData.typeSearchable, // SEARCHABLE
+                                true, // UNSIGNED_ATTRIBUTE
+                                false, // FIXED_PREC_SCALE
+                                false, // AUTO_INCREMENT
+                                null, // LOCAL_TYPE_NAME
+                                0, // MINIMUM_SCALE
+                                0, // MAXIMUM_SCALE
+                                null, // SQL_DATA_TYPE (unused)
+                                null, // SQL_DATETIME_SUB (unused)
+                                BASE_10 // NUM_PREC_RADIX
+                        )),
+                        new ArrayList<>(Arrays.asList(
+                                JdbcType.BINARY.name(), // TYPE_NAME
+                                JdbcType.BINARY.getJdbcType(), // DATA_TYPE
+                                TYPE_COLUMN_SIZE_MAP.get(JdbcType.BINARY), // PRECISION
+                                "x'", // LITERAL_PREFIX
+                                "'", // LITERAL_SUFFIX
```

Comment:
We do not have the same syntax as SQL Server since we use Calcite to parse the SQL. I think we should follow Calcite's implementation for the types we support which does report to use x '' for literals and this is what we have tested and documented to work w/ the driver as well (https://github.com/apache/calcite/blob/7e38304093d803fba678817c07012f4d201ad8a0/core/src/main/java/org/apache/calcite/jdbc/CalciteMetaImpl.java#L412 
https://github.com/apache/calcite/blob/7e38304093d803fba678817c07012f4d201ad8a0/core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystemImpl.java#L159 )

Reply:
ok I see. Thanks for your reply! @andiem-bq That makes sense and I'll resolve this comment.